src/standalone/Vegetation/Canopy.jl

```diff
@@ -367,14 +368,15 @@ function ClimaLSM.make_update_aux(
         q::FT = canopy.atmos.q(t)
 
         # update radiative transfer
-        PAR .= compute_PAR(canopy.radiative_transfer, canopy.radiation, t)
         K = extinction_coeff(ld, θs)
+        PAR .= compute_PAR(canopy.radiative_transfer, canopy.radiation, t)
         @. APAR = plant_absorbed_ppfd(
```

Comment:
I think it would be better to have multiple methods for plant_absorbed_ppfd based on the type of canopy.radiative_transfer, rather than the # of arguments. the parameters can be unpack within the function, then?

Reply:
I wanted to do this initially, but there is an issue evaluating that way. Since plant_absorbed_ppfd is called element-wise using "@." to unpack the PAR argument which is stored as a field vector, I couldn't directly pass the radiation model struct or parameters struct without throwing an error from calling @. on the struct. This was the work around I found for this but it is messy so if there is a better way I would love to know.

Reply:
ah, yes! youre right. There are a couple ways around that...
Take a look at the function `matric_potential` here:
https://github.com/CliMA/ClimaLSM.jl/blob/bccd7217ec7ba45e820c616253af156d177b2ff8/src/standalone/Soil/soil_hydrology_parameterizations.jl#L40

which dispatches off of the type of the first argument. Here is another method for it:
https://github.com/CliMA/ClimaLSM.jl/blob/bccd7217ec7ba45e820c616253af156d177b2ff8/src/standalone/Soil/soil_hydrology_parameterizations.jl#L140

the first option is to do this:
`p.soil.psi .= matric_potential.(Ref(hcm), S) `
where `S` is a field or array and `hcm` is the struct. `Ref` effectively makes it so you can broadcast the struct as well. The downside to this is that we now need two broadcasting calls, one in the `.=` and one in the function itself. doing this:
`@. p.soil.psi = matric_potential(Ref(hcm), S) ` which would reduce it to a single broadcast is more efficient, but it will fail because it will also try and broadcast the function `Ref`.

Second option:
Make your struct broadcastable....
https://github.com/CliMA/ClimaLSM.jl/blob/bccd7217ec7ba45e820c616253af156d177b2ff8/src/standalone/Soil/retention_models.jl#L18

then you can do this:
`@. p.soil.psi = matric_potential(hcm, S)`

I think we will move towards this for most of our structs, see:https://github.com/CliMA/ClimaLSM.jl/issues/209

Reply:
Got it! Cleaned it up using the second approach you suggested. Thanks

Reply:
nice! glad it worked